### PR TITLE
Fix - multiple overwrite may reference to same invalid link.

### PR DESCRIPTION
**Pull request by hellosnow**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Common/Reference/ApplyOverwriteDocument.cs [hunk 1]
```diff
@@ -94,8 +94,12 @@ select new
                 {
                     pair.model.LinkToUids = pair.model.LinkToUids.Union(overwriteDocumentModel.LinkToUids);
                     pair.model.LinkToFiles = pair.model.LinkToFiles.Union(overwriteDocumentModel.LinkToFiles);
-                    pair.model.FileLinkSources = pair.model.FileLinkSources.Union(overwriteDocumentModel.FileLinkSources.ToImmutableDictionary(k => k.Key, k => k.Value.ToImmutableList())).ToImmutableDictionary();
-                    pair.model.UidLinkSources = pair.model.UidLinkSources.Union(overwriteDocumentModel.UidLinkSources.ToImmutableDictionary(k => k.Key, k => k.Value.ToImmutableList())).ToImmutableDictionary();
+                    pair.model.FileLinkSources = pair.model.FileLinkSources.Union(
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Common/Reference/ApplyOverwriteDocument.cs]: Use CollectionUtility.Merge as similar to BuildManagedReferenceDocument?

Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Common/Reference/ApplyOverwriteDocument.cs]: I just added new method to suitable for IImmutableDictionary

---
In reply to: [100939213](https://github.com/dotnet/docfx/pull/1298#discussion_r100939213) [](ancestors = 100939213)

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Common/Reference/ApplyOverwriteDocument.cs (hunk 1 above)
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Common/Reference/ApplyOverwriteDocument.cs]: offline disucss and test with lianwei, reuse the original method, which has better perf.

---
In reply to: [100957089](https://github.com/dotnet/docfx/pull/1298#discussion_r100957089) [](ancestors = 100957089,100939213)

**Pull request merged**